cheroot/test/test_core.py

```diff
@@ -283,7 +279,6 @@ def test_content_length_required(test_client):
 
 
 @pytest.mark.xfail(
-    IS_CIRCLE_CI_ENV,
```

Comment:
with the test changes in this PR, i found this test was failing reliability locally for me as well, rather than just on circleci - i noticed this in response to the relocation of the `wsgi_server` and `native_server` fixtures above.